### ISSUE TITLE
Update Statistics on Partitioned Table - getting 'NORECOMPUTE' is not a recognized 

### ISSUE THREAD

**Issue by ShakerGit** [olahallengren/sql-server-maintenance-solution#285]
**Description of the issue**
Getting 'NORECOMPUTE' is not a recognized UPDATE STATISTICS option when running IndexOptimize on partitioned Table. 
UPDATE STATISTICS [xyzdb].[dbo].[xyzTBL] [_WA_Sys_00000005_206A9855] WITH RESAMPLE, NORECOMPUTE ON PARTITIONS(1)

If I manually run the command switching the NORECOMPUTE and RESAMPLE options as below then it runs fine.
UPDATE STATISTICS [xyzdb].[dbo].[xyzTBL] [_WA_Sys_00000005_206A9855] WITH NORECOMPUTE, RESAMPLE ON PARTITIONS(1)

**SQL Server version and edition**
Execute ```SELECT @@VERSION```
Microsoft SQL Server 2017 (RTM-CU16) (KB4508218) - 14.0.3223.3 (X64)   Jul 12 2019 17:43:08   Copyright (C) 2017 Microsoft Corporation  Enterprise Edition (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) 

**Version of the script**
Check the header of the stored procedure
Version: 2019-06-14 00:05:34

**What command are you executing?**
IndexOptimize
UPDATE STATISTICS [xyzdb].[dbo].[xyzTBL] [_WA_Sys_00000005_206A9855] WITH RESAMPLE, NORECOMPUTE ON PARTITIONS(1)

**What output are you getting?**
Msg Msg 155, 'NORECOMPUTE' is not a recognized UPDATE STATISTICS option.

**Comment by ShakerGit** [comment]
For now, changed the IndexOptimize to Insert the NORECOMPUTE argument before the RESAMPLE argument.

         IF @CurrentNoRecompute = 1
          BEGIN
            INSERT INTO @CurrentUpdateStatisticsWithClauseArguments (Argument)
            SELECT 'NORECOMPUTE'
          END

          IF @CurrentStatisticsResample = 'Y'
          BEGIN
            INSERT INTO @CurrentUpdateStatisticsWithClauseArguments (Argument)
            SELECT 'RESAMPLE'
          END

**Comment by ShakerGit** [comment]
Sorry, meant to just add the comment, not close the issue.

**Comment by codykonior** [comment]
I've also experienced this issue.

**Comment by dbaduff** [comment]
This seems like such an easy fix. Why hasn't it just been added to the official release?

**Comment by codykonior** [comment]
I don’t want to speak for him but writing free software is unpaid work that never ends, making changes that he doesn’t need.

It would be nice to have one source but we really can’t demand anything.

**Comment by dbaduff** [comment]
You are right of course. I am not trying to criticize Ola or what he chooses to focus on. Just observing that it is an easy fix. Of course, There may be ramifications when testing against all versions and environments so I should just keep quiet. I'm glad I have a solution.

**Comment by codykonior** [comment]
I had the same thoughts. My reply is as much to myself as it is to you 😅

**Comment by nanoDBA** [comment]
Changing IndexOptimize to Insert the NORECOMPUTE argument before the RESAMPLE argument worked for me as well per @ShakerGit 's suggestion in https://github.com/olahallengren/sql-server-maintenance-solution/issues/285#issuecomment-619363348

**Comment by EitanBlumin** [comment]
I just had the same error and found this thread.
My friends, isn't this an open-source project?

If you already know how to fix this, why not just open a pull request with the fix?

**Comment by olahallengren** [comment]
This issue has been fixed in https://github.com/olahallengren/sql-server-maintenance-solution/pull/872. Thank you for reporting it.
